Here are some tweaks to your webvtt-editor

lenniea opened the pull request:
Converted app.js -> app.ts (Typescript)
Fixed formatting of video timestamps to include msec (per WebVTT spec)
Added (optional) positioning in top-left, top-right, bottom-left, bottom-right corners for a "score board"
Simplified UI to reduce horizontal size & increased size of video preview

lenniea committed:
Fix WEBVTT timestamp format on Export (hh:mm:ss.xxx)

lenniea committed:
Support subtitle "Scoreboards" in top-left, top-right, bot-left, bot-right corners
Fix sorting of equal start times
Add "#pos" select option in index.html

lenniea committed:
Simplify <table></table> HTML
Increase size=5

lenniea committed:
Remove console.log debug
Increase width of main-left

lenniea committed:
Cleanup Javascript for readability

lenniea committed:
Convert app.js to typescript
(couldn't figure out how to compile sw.js with webworker!)

lenniea committed:
Add package-lock.json & package.json

lenniea committed:
Update .gitignore

lenniea committed:
Add screenshot & fix typos in README.md

lenniea committed:
Update README.md

opusonline commented:
Hi! Thanks for your updates! It's always welcome!
I had a look and noticed in [package.json](https://github.com/lenniea/webvtt-editor/blob/master/package.json) I can see a dependency to jquery, but I couldn't find its usage in the app. If jquery is not needed, please remove it. Thanks!

lenniea committed:
Add buttons for score keeping (only active if #posOption != None
Pad Player/Team names to 30 characters if #posOption != None
Add test/Scoreboard.vtt for testing

lenniea commented:
> Hi! Thanks for your updates! It's always welcome! I had a look and noticed in [package.json](https://github.com/lenniea/webvtt-editor/blob/master/package.json) I can see a dependency to jquery, but I couldn't find its usage in the app. If jquery is not needed, please remove it. Thanks!

I'm not much of a Javascript expert but I believe its used [indirectly] by the $( ) in lines 66-85???  Also added some more buttons to automate scorekeeping ;-) as well as a test/Scoreboard.vtt file for testing purposes!

lenniea commented:
Here's my commits from yesterday & today - hope you like them and merge them into your main branch.  Thanks for the initial post I used it to add subtitles to a youtube pickleball video on my channel :-)

https://www.youtube.com/watch?v=A0UkYuviikw

lenniea committed:
Add year+name to LICENSE.md

lenniea committed:
Update LICENSE.md

Add nl for better formatting

lenniea commented:
Did some more minor cleanups

lenniea committed:
Add utility functions round, min, max, padLeft, padRight, trimLeft
Change var -> let
Add logic to support Pickleball "serves" (2) or Volleyball/Basketball "sideout/poss"
Use padLeft to improve scoreboard formatting
Translate alerts from German to English

lenniea committed:
Change hover/title to "side out/poss"

lenniea committed:
Fix formatTableRow bug introduced by refactoring

lenniea committed:
Handle single line scores (e.g. 0-0-2)
Don't allow scores to go negative
Improve index.html layout/text

lenniea committed:
Pause video & add logic for updateLastCue for basketball (1,2,3 point) scoring
Abbreviate Bottom as Bot for more compact dropdown

lenniea committed:
Add Basketball/Pickleball/Volleyball.vtt files for testing

lenniea committed:
Support dynamic scoreboard width based on name length
Replace trimLeft() by built-in trim()
Cleanup Scoreboard.vtt

lenniea committed:
Fix typo in refactoring & minor improvements

lenniea committed:
Fix sorting caused by refactoring
Simplify compCues

lenniea committed:
Increase width & size of scoreboards
